https://issues.redhat.com/browse/ACM-7118 (#5748)
open-cluster-management/rhacm-docs: observability/customize_observability.adoc

This secret can be mounted into all components that need to access the object store, and it includes the following components: `receiver`, `store`, `ruler`, `compact`.

[#configuring-proxy-settings-for-observability-add-ons]
== Configuring proxy settings for observability add-ons

Configure the proxy settings to allow the communications from the managed cluster to access the hub cluster through a HTTP and HTTPS proxy server. Typically, add-ons do not need any special configuration to support HTTP and HTTPS proxy servers between a hub cluster and a managed cluster. But if you enabled the observability add-on, you must complete the following configuration: 

=== Prerequisite 

> - You have a hub cluster. 

Reviewer: ```suggestion

- You have a hub cluster. 
```

- You have enabled the proxy settings between the hub cluster and managed cluster. 

. Go to the cluster namespace on your hub cluster. 

> . Create an `AddOnDeploymentConfig` with the proxy settings by adding a `spec.proxy.Config`. View the following YAML example:

Reviewer: ```suggestion
. Create an `AddOnDeploymentConfig` resource with the proxy settings by adding a `spec.proxyConfig` parameter. View the following YAML example:
```

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>
  namespace: <managed-cluster-name>
spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:
    httpsProxy: "https://<username>:<password>@<ip>:<port>" <1>
    noProxy: ".cluster.local,.svc,172.30.0.1" <2> 
----
+

> <1> Specify the same name for the HTTP and HTTPS proxy.

Reviewer 2: > <1> Specify the same name for the HTTP and HTTPS proxy.

It not clear here. What I want to say here is you are able to specify either a HTTP proxy (http://<username>:<password>@<ip>:<port>) or a HTTPS proxy (https://<username>:<password>@<ip>:<port>) for this field.

> <2> Include the IP address of the `kube-apiserver`. To get the IP address, run following command on your managed cluster: `oc -n default describe svc kubernetes | grep IP:`. 
> . Go to the `ManagedClusterAddOn` and reference the `AddOnDeploymentConfig` you made by adding an item in `spec.configs`. View the following YAML example: 
> 
> [source,yaml]

Reviewer: Should these steps be a part of the YAML explanation or are the steps a part of the configuring procedure? 

```suggestion
<2> Include the IP address of the `kube-apiserver`. 

. To get the IP address, run following command on your managed cluster: 
+
[source,bash]
----
oc -n default describe svc kubernetes | grep IP:
----
 
. Go to the `ManagedClusterAddOn` resource and reference the `AddOnDeploymentConfig` you made by adding an item in `spec.configs`. View the following YAML example: 
+
[source,yaml]
```

----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>

>   namespace: <managed-cluster-name>

Reviewer 2: This is not the yaml of the sample `ManagedClusterAddOn`.

spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:
    httpsProxy: "https://<username>:<password>@<ip>:<port>"
    noProxy: ".cluster.local,.svc,172.30.0.1"
----

[#disabling-proxy-settings-for-observability-add-ons]
== Disabling proxy settings for Observability add-ons

> If your development needs change, you might need to disable the Observability add-ons you configured for the hub cluster and managed cluster. You can disable the Observability add-on at any time. ,

Reviewer 2: Here we're going to disable the proxy settings for the Observability add-on, not to disable the add-on itself.

This secret can be mounted into all components that need to access the object store, and it includes the following components: `receiver`, `store`, `ruler`, `compact`.

[#configuring-proxy-settings-for-observability-add-ons]
== Configuring proxy settings for observability add-ons

> Configure the proxy settings to allow the communications from the managed cluster to access the hub cluster through a HTTP and HTTPS proxy server. Typically, add-ons do not need any special configuration to support HTTP and HTTPS proxy servers between a hub cluster and a managed cluster. But if you enabled the observability add-on, you must complete the following configuration: 

Reviewer: ```suggestion
Configure the proxy settings to allow the communications from the managed cluster to access the hub cluster through a HTTP and HTTPS proxy server. Typically, add-ons do not need any special configuration to support HTTP and HTTPS proxy servers between a hub cluster and a managed cluster. But if you enabled the observability add-on, you must complete the proxy configuration. 
```

> === Prerequisite 

Reviewer: ```suggestion
== Prerequisite 
```

- You have a hub cluster. 
- You have enabled the proxy settings between the hub cluster and managed cluster. 

> . Go to the cluster namespace on your hub cluster. 

Reviewer: ```suggestion
Complete the following steps to configure the proxy settings for the observability add-on:

. Go to the cluster namespace on your hub cluster. 
```

. Create an `AddOnDeploymentConfig` resource with the proxy settings by adding a `spec.proxyConfig` parameter. View the following YAML example:
+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>
  namespace: <managed-cluster-name>
spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:
    httpsProxy: "https://<username>:<password>@<ip>:<port>" <1>
    noProxy: ".cluster.local,.svc,172.30.0.1" <2> 
----
+

> <1> For this field, you can specify either a HTTP proxy, `http://:@:`, or a HTTPS proxy, `https://:@:`.

Reviewer 2: > <1> For this field, you can specify either a HTTP proxy, `http://:@:`, or a HTTPS proxy, `https://:@:`.

Change to

> <1> For this field, you can specify either a HTTP proxy or a HTTPS proxy.

<2> Include the IP address of the `kube-apiserver`. 

. To get the IP address, run following command on your managed cluster: 
+
[source,bash]
----
oc -n default describe svc kubernetes | grep IP:
----
 

> . Go to the `ManagedClusterAddOn` resource and reference the `AddOnDeploymentConfig` you made by adding an item in `spec.configs`. View the following YAML example: 

Reviewer: ```suggestion
. Go to the `ManagedClusterAddOn` resource and update it by referencing the `AddOnDeploymentConfig` resource that you made. View the following YAML example:

+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: ManagedClusterAddOn
metadata:
  name: observability-controller
  namespace: <managed-cluster-name>
spec:
  installNamespace: open-cluster-managment-addon-observability
  configs:
  - group: addon.open-cluster-management.io

>     resource: addondeploymentconfigs

Reviewer: Should the resource match the `kind` of the configuration? 
```suggestion
    resource: AddonDeploymentConfig
```

    name: <addon-deploy-config-name>
    namespace: <managed-cluster-name> 
----

> 

Reviewer: How can the user verify that the proxy settings are configured correctly? Do they have to attempt to communicate to the hub cluster from the managed cluster ?

Author: @elgnay may I please get your insight and help here to answer Mikela's question on verification?

Reviewer 2: If the proxy settings are configured correctly, the metric collector deployed by the obs add-on agent on the managed cluster will send metrics data to the hub. And then user is able to view the metrics from that managed cluster on the Grafana dashboard from the hub cluster.

[#disabling-proxy-settings-for-observability-add-ons]
== Disabling proxy settings for observability add-ons

> If your development needs change, you might need to disable the proxy setting for the observability add-ons you configured for the hub cluster and managed cluster. You can disable the proxy settings for the observability add-on at any time. 

Reviewer: ```suggestion
If your development needs change, you might need to disable the proxy setting for the observability add-ons you configured for the hub cluster and managed cluster. You can disable the proxy settings for the observability add-on at any time. Complete the following steps:
```

> . Go to the `ManagedClusterAddOn`.

Reviewer: ```suggestion
. Go to the `ManagedClusterAddOn` resource.
```

> . Remove the referenced `AddOnDeploymentConfig`. 

Reviewer: Should the entire `config.group` be removed or only the `resource` contents?
```suggestion
. Remove the referenced `AddOnDeploymentConfig` resource.
```

[#configuring-proxy-settings-for-observability-add-ons]
== Configuring proxy settings for observability add-ons

Configure the proxy settings to allow the communications from the managed cluster to access the hub cluster through a HTTP and HTTPS proxy server. Typically, add-ons do not need any special configuration to support HTTP and HTTPS proxy servers between a hub cluster and a managed cluster. But if you enabled the observability add-on, you must complete the proxy configuration. 

== Prerequisite 

- You have a hub cluster. 
- You have enabled the proxy settings between the hub cluster and managed cluster. 

Complete the following steps to configure the proxy settings for the observability add-on:

. Go to the cluster namespace on your hub cluster. 
. Create an `AddOnDeploymentConfig` resource with the proxy settings by adding a `spec.proxyConfig` parameter. View the following YAML example:
+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>
  namespace: <managed-cluster-name>
spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:

>     httpsProxy: "https://<username>:<password>@<ip>:<port>" <1>

Reviewer 2: Considering the HTTPS proxy is not supported in this release, it's better to use `http://<username>:<password>@<ip>:<port>` instead of `https://<username>:<password>@<ip>:<port>` here.

kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>
  namespace: <managed-cluster-name>
spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:
    httpsProxy: "http://<username>:<password>@<ip>:<port>" <1>
    noProxy: ".cluster.local,.svc,172.30.0.1" <2> 
----
+
<1> For this field, you can specify either a HTTP proxy or a HTTPS proxy.
<2> Include the IP address of the `kube-apiserver`. 

. To get the IP address, run following command on your managed cluster: 
+
[source,bash]
----
oc -n default describe svc kubernetes | grep IP:
----
 
. Go to the `ManagedClusterAddOn` resource and update it by referencing the `AddOnDeploymentConfig` resource that you made. View the following YAML example:
+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: ManagedClusterAddOn
metadata:
  name: observability-controller
  namespace: <managed-cluster-name>
spec:
  installNamespace: open-cluster-managment-addon-observability
  configs:
  - group: addon.open-cluster-management.io
    resource: AddonDeploymentConfig
    name: <addon-deploy-config-name>
    namespace: <managed-cluster-name> 
----

> . Verify the proxy settings. If you successfully configured the proxy settings, the metric collector deployed by the observability add-on agent on the managed cluster sends the data to the hub cluster.  Complete the following steps:

Reviewer: ```suggestion
. Verify the proxy settings. If you successfully configured the proxy settings, the metric collector deployed by the observability add-on agent on the managed cluster sends the data to the hub cluster. Complete the following steps:
```

> . Go to the hub cluster then the managed cluster on the Grafana dashboard. 
> . View the metrics for the proxy settings. 

Reviewer: Creating substeps
```suggestion
.. Go to the hub cluster then the managed cluster on the Grafana dashboard. 
.. View the metrics for the proxy settings. 
```

kind: AddOnDeploymentConfig
metadata:
  name: <addon-deploy-config-name>
  namespace: <managed-cluster-name>
spec:
  agentInstallNamespace: open-cluster-managment-addon-observability
  proxyConfig:
    httpsProxy: "http://<username>:<password>@<ip>:<port>" <1>
    noProxy: ".cluster.local,.svc,172.30.0.1" <2> 
----
+
<1> For this field, you can specify either a HTTP proxy or a HTTPS proxy.
<2> Include the IP address of the `kube-apiserver`. 

. To get the IP address, run following command on your managed cluster: 
+
[source,bash]
----
oc -n default describe svc kubernetes | grep IP:
----
 
. Go to the `ManagedClusterAddOn` resource and update it by referencing the `AddOnDeploymentConfig` resource that you made. View the following YAML example:
+
[source,yaml]
----
apiVersion: addon.open-cluster-management.io/v1alpha1
kind: ManagedClusterAddOn
metadata:
  name: observability-controller
  namespace: <managed-cluster-name>
spec:
  installNamespace: open-cluster-managment-addon-observability
  configs:
  - group: addon.open-cluster-management.io
    resource: AddonDeploymentConfig
    name: <addon-deploy-config-name>
    namespace: <managed-cluster-name> 
----

> If you successfully configured the proxy settings, the metric collector deployed by the observability add-on agent on the managed cluster sends the data to the hub cluster.  

Author: Reviewers, please let me know what you think of the new verification section.

> If you successfully configured the proxy settings, the metric collector deployed by the observability add-on agent on the managed cluster sends the data to the hub cluster.  
> 
> Complete the following steps to verify the proxy settings:

Reviewer: I would make this a step

```suggestion
. Verify the proxy settings. If you successfully configured the proxy settings, the metric collector deployed by the observability add-on agent on the managed cluster sends the data to the hub cluster.  Complete the following steps:
```
